java/vortex-jni/build.gradle.kts

```diff
@@ -0,0 +1,94 @@
+plugins {
+    `java-library`
+}
+
+dependencies {
+    api("net.java.dev.jna:jna-platform")
+    errorprone("com.google.errorprone:error_prone_core")
+    errorprone("com.jakewharton.nopen:nopen-checker")
+
+    implementation("com.google.guava:guava")
+    compileOnly("com.google.errorprone:error_prone_annotations")
+    compileOnly("com.jakewharton.nopen:nopen-annotations")
+
+    testImplementation("org.junit.jupiter:junit-jupiter")
+}
+
+val vortexFFI = projectDir.parentFile.parentFile.resolve("vortex-ffi")
+
+val platformLibSuffix =
+    if (System.getProperty("os.name").contains("Mac")) {
+        "dylib"
+    } else {
+        "so"
+    }
+
+val targetDir = projectDir.parentFile.parentFile.resolve("target")
+val libraryFile = targetDir.resolve("release/libvortex_ffi.$platformLibSuffix")
+
+val cargoCheck by tasks.registering(Exec::class) {
+    workingDir = vortexFFI
+    commandLine("cargo", "check")
+}
+
+val cargoBuild by tasks.registering(Exec::class) {
+    workingDir = vortexFFI
+    commandLine(
+        "cargo",
+        "build",
+        "--release",
+    )
+
+    outputs.files(libraryFile)
+
+    // Always force rebuilds, rely on cargo's builtin caching and incremental compile to avoid spurious rebuilds.
+    outputs.upToDateWhen { false }
+    outputs.cacheIf { false }
+
+    dependsOn.add(cargoCheck)
```

Comment:
It seems mildly annoying to cargo check every time I build?

Reply:
You should hang it off the builtin `check` task

Reply:
this is done